packages/store/addon/-private/system/core-store.ts

```diff
@@ -3655,3 +3657,43 @@ if (DEBUG) {
     }
   };
 }
+
+/**
+ * Flag indicating whether all inverse records are available
+ *
+ * true if the inverse exists and is loaded (not empty)
+ * true if there is no inverse
+ * false if the inverse exists and is not loaded (empty)
+ *
+ * @return {boolean}
+ */
+function areAllInverseRecordsLoaded(store: CoreStore, resource: JsonApiRelationship): boolean {
+  const cache = identifierCacheFor(store);
+
+  if (Array.isArray(resource.data)) {
+    // treat as collection
+    // check for unloaded records
+    let hasEmptyRecords = resource.data.reduce((hasEmptyModel, i) => {
```

Comment:
@runspired should we rename `i` to `resourceIdentifier`?

Reply:
seems like a good rename